### PR TITLE
RCL topic update for 3.0

### LINKED ISSUE #14408: RCL topic update for 3.0
Call out -support-pages-and-views option when creating an RCL

> The Razor class library (RCL) template defaults to Razor component development by default. A new template option in Visual Studio provides template support for pages and views. When creating an RCL from the template in a command shell, pass the `-support-pages-and-views` option (`dotnet new razorclasslib -support-pages-and-views`).

This is a 3.0 update. https://github.com/aspnet/AspNetCore/issues/7536

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 9fd019c7-ceb7-6ef3-26ba-6b44341b6fc5
* Version Independent ID: fd116d00-0dc7-efaf-c714-34d77b345a05
* Content: [Reusable Razor UI in class libraries with ASP.NET Core](https://docs.microsoft.com/en-us/aspnet/core/razor-pages/ui-class?view=aspnetcore-2.2&tabs=visual-studio)
* Content Source: [aspnetcore/razor-pages/ui-class.md](https://github.com/aspnet/AspNetCore.Docs/blob/master/aspnetcore/razor-pages/ui-class.md)
* Product: **aspnet-core**
* Technology: **aspnetcore-razorpages**
* GitHub Login: @Rick-Anderson
* Microsoft Alias: **riande**

### PR DESCRIPTION
**Pull request by guardrex**
Fixes #14408

[Internal Review Topic](https://review.docs.microsoft.com/en-us/aspnet/core/razor-pages/ui-class?view=aspnetcore-2.1&branch=pr-en-us-14461)

* Whole-topic versioning.
* Add info on "Pages and Views" via VS and `-support-pages-and-views` option.&dagger;
* Update sample code for 2.x to 2.2 from 2.1.

&dagger;I won't get a look at the VS screens locally until RTM (I can't support preview VS).

Options:

* Let me know how to phrase the VS UI content ... How and where does VS show 'pages and views support'? Send me a screenshot perhaps.
* Let's hold this PR until RTM ... My VS will update that day ... I'll see and react to it myself. I'll just mark this PR with `[HOLD]` in the title for now.

Also note: I thought that I would arrive to find this topic+sample code addressed for 3.0. *Not that lucky.* Some 3.0 content for RCL is here, but the walkthrough is 2.x. Therefore, I take that walkthrough out for the 3.0 version of the topic (recommended anyway btw ... would be better in a proper tutorial imo). I wasn't anticipating this scenario. I'm 🏃😅 right now; so if you prefer a 3.0 walkthrough, I recommend that we open a new issue for this topic (or for a separate tutorial) and I or someone else can work it later. I need to get back to my 3.0 high priority issues. 🏃🏃🏃

### CODE REVIEW

File: aspnetcore/razor-pages/ui-class.md [hunk 1]
```diff
@@ -4,14 +4,184 @@ author: Rick-Anderson
...
+[View or download sample code](https://github.com/aspnet/AspNetCore.Docs/tree/master/aspnetcore/razor-pages/ui-class/samples) ([how to download](xref:index#how-to-download-a-sample))
+
+## Create a class library containing Razor UI
+
+# [Visual Studio](#tab/visual-studio)
+
+* From the Visual Studio **File** menu, select **New** > **Project**.
+* Select **ASP.NET Core Web Application**.
+* Name the library (for example, "RazorClassLib") > **OK**. To avoid a file name collision with the generated view library, ensure the library name doesn't end in `.Views`.
+* Verify **ASP.NET Core 2.1** or later is selected.
```
Comment (Rick-Anderson) [inline aspnetcore/razor-pages/ui-class.md]: 3.0

File: aspnetcore/razor-pages/ui-class.md [hunk 2]
```diff
@@ -4,14 +4,184 @@ author: Rick-Anderson
...
 ---
 # Create reusable UI using the Razor class library project in ASP.NET Core
 
 By [Rick Anderson](https://twitter.com/RickAndMSFT)
 
+::: moniker range=">= aspnetcore-3.0"
+
+Razor views, pages, controllers, page models, [Razor components](xref:blazor/class-libraries), [View components](xref:mvc/views/view-components), and data models can be built into a Razor class library (RCL). The RCL can be packaged and reused. Applications can include the RCL and override the views and pages it contains. When a view, partial view, or Razor Page is found in both the web app and the RCL, the Razor markup (*.cshtml* file) in the web app takes precedence.
+
+This feature requires [!INCLUDE[](~/includes/2.1-SDK.md)]
```
Comment (Rick-Anderson) [inline aspnetcore/razor-pages/ui-class.md]: 3.0 ?

Comment (Rick-Anderson) [reply aspnetcore/razor-pages/ui-class.md]: ## Use the Razor UI library from a Razor Pages project

Is missing from the 3.0 version.

**Comment by guardrex** [comment]
There was more than that ... a 3.0 section at the bottom. I'm also cutting out the walkthru from the 3.0 content. It would flip in-and-out of versioning a half-dozen times to get all of these bits in.

**Comment by Rick-Anderson** [comment]
Whole topic versions is the right choice.

Why did the walkthrough get cut?

image

**Comment by guardrex** [comment]
That section you called out is part of the walkthru.

Cut because it would require a complete run-thru for 3.0 (e.g., VS deltas) with 3.0 samples (project + RCL). Recommend that we open a new issue to come back later and either add a 3.0 walkthru here with all the trim'ins or create a separate tutorial doc in the tutorials area.

Another factor here is that I can't do the update without either an Aquent laptop, which I don't have, or waiting for VS to RTM with the 3.0-era bits. Getting them to send a laptop at this late date doesn't make sense, so we'd at least need to hold this if you want to wait ... I mean if just opening a new issue isn't your choice.

**Comment by Rick-Anderson** [comment]
Create an issue and assign to me.

**Pull request merged**